vars/deployProd.groovy

```diff
@@ -35,7 +35,46 @@ def call(body) {
 
 
   stage('Checkout Tag') {
-    sh "git checkout tags/${config.prodVersion}"
+    sh "git checkout master"
+  }
+
+  def url = ''
+  def urlMinusProtocol = ''
+  withCredentials([usernamePassword(credentialsId: 'github', passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+      stage('Check master branch') {
+          url = sh(returnStdout: true, script: 'git config remote.origin.url').trim()
+          urlMinusProtocol = url.substring(url.indexOf('://')+3)
+
+          sh "git fetch --no-tags --progress https://${GIT_USERNAME}:${GIT_PASSWORD}@${urlMinusProtocol} +refs/tags/${config.prodVersion}:refs/remotes/origin/master"
+
+          //Compare to master branch to look for any unmerged changes in the tag
+          def commitsBehind = sh(returnStdout: true, script: "git rev-list --right-only --count refs/tags/${config.prodVersion}...remotes/origin/master").trim().toInteger()
+          if (commitsBehind > 0) {
+              error("Master Branch has changesets not included in this tag ${config.prodVersion}. Please merge master into the tag before deploying to prod.")
+          } else {
+              echo "Tag ${config.prodVersion} is up to date with changesets on master. Proceeding with release..."
+          }
+
+          //Let's do the same thing with master too, just in case
+          def commitsBehind = sh(returnStdout: true, script: "git rev-list --right-only --count refs/heads/master...remotes/origin/master").trim().toInteger()
```

Comment:
Not sure of the reason to do this since you just checked out the master branch seconds ago.